### DISCUSSION TITLE
Publish outbox metrics ?

### DISCUSSION THREAD

**Discussion by ironhide0396** [gruelbox/transaction-outbox#675] [Ideas]
While the library aims to process the outbox entry immediately, in case of certain heavy writes and sequencing involved, we can get into a condition where it is not necessary that the node which published the event will be the same one processing it.
It would be wise to publish metrics from the library in such cases so that in case of extreme writes, an end user can be alerted and maybe spawn up a few more replicas to process the load and drain the outbox.

If this sounds something useful, I can chalk out an PR.

**Comment by ironhide0396** [comment]
Or do you believe that overriding the listeners are enough for every consumer of this library ?

**Comment by badgerwithagun** [comment]
I wonder whether this is an extra method in the listener, e.g  `onProcessingDeferred(...)`?

Fired when sequential processing is used _or_ execution is rejected by the `Submitter`.
